packages/ckeditor5-image/src/imageresize/imageresizebuttons.ts

```diff
@@ -39,7 +43,7 @@ export default class ImageResizeButtons extends Plugin {
 	 * @inheritDoc
 	 */
 	public static get requires() {
-		return [ ImageResizeEditing ] as const;
+		return [ ImageResizeEditing, ImageCustomResizeUI ] as const;
```

Comment:
Is that requirement necessary? I mean, we could soft-detect whether the plugin is loaded and then add a button or not. This just feels like an unnecessary feature coupling.

Reply:
@oleq I'm not sure if it is needed. We already have `isCustomImageResizeOption` which detects if custom dropdown entry belongs to `ImageCustomResizeUI` plugin (and creates dropdown entry for such plugin) so plugins are coupled anyway. It would make sense if `ImageCustomResizeUI` had possibility to dynamically append dropdown entries to `ImageResizeButtons` plugin but on the other hand it'd be more complicated and probably noone would even use these two plugins separately because menu entries are configured via config.

Reply:
@oleq I made this dependency optional https://github.com/ckeditor/ckeditor5/pull/16140/commits/b24625433d2cb4bfa62a21e7312a6f316736e99b. However, I kept the custom button in the dropdown because it can be used in the future in third party custom integrations.